src/transactions/PayForFiles.ts

```diff
@@ -0,0 +1,41 @@
+import { Transaction } from './Transaction';
+import {
+  PublicKey,
+  SystemProgram,
+  TransactionCtorFields,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import { MemoProgram } from './programs';
+
+type PayForFilesParams = {
+  lamports: number;
+  fileHashes: Buffer[];
+  arweaveWallet?: PublicKey;
+};
+
+export class PayForFiles extends Transaction {
+  constructor(options: TransactionCtorFields, params: PayForFilesParams) {
+    const { feePayer } = options;
+    const { lamports, fileHashes, arweaveWallet } = params;
+
+    super(options);
+
+    this.add(
+      SystemProgram.transfer({
+        fromPubkey: feePayer,
+        toPubkey: arweaveWallet ?? new PublicKey('HvwC9QSAzvGXhhVrgPmauVwFWcYZhne3hVot9EbHuFTm'),
```

Comment:
Should this go to some sort of build-time environment variable? Feels wrong to just hardcode it :)